packages/eslint-plugin/src/rules/sort-ngmodule-metadata-arrays.ts

```diff
@@ -51,18 +52,26 @@ export default createESLintRule<Options, MessageIds>({
           ) {
             return;
           }
-          const unorderedNode = initializer.elements
+          const unorderedNodes = initializer.elements
             .filter(isIdentifier)
-            .find(({ name }, index, list) => {
-              const nextElementName = list[index + 1]?.name;
-              return (
-                nextElementName && name.localeCompare(nextElementName) === 1
-              );
+            .map((current, index, list) => {
+              return [current, list[index + 1]];
+            })
+            .find(([current, next]) => {
+              return next && current.name.localeCompare(next?.name) === 1;
```

Comment:
Since you're checking the truthiness of `next` previously we can remove the optional chaining here:

```suggestion
              return next && current.name.localeCompare(next.name) === 1;
```